src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpointIT.java

```diff
@@ -0,0 +1,150 @@
+     */
+    @Test
+    public void shouldCreateCollectionExercise() throws CTPException {
+        String exerciseRef = "899990";
+        String userDescription = "Test Description";
+        Pair<Integer, String> result = this.client.createCollectionExercise(TEST_SURVEY_ID, exerciseRef,
+                userDescription);
+
+        assertEquals(201, (int) result.getLeft());
+
+        CollectionExerciseDTO newCollex = this.client.getCollectionExercise(result.getRight());
+
+        assertEquals(TEST_SURVEY_ID, UUID.fromString(newCollex.getSurveyId()));
+        assertEquals(exerciseRef, newCollex.getExerciseRef());
+        assertEquals(userDescription, newCollex.getUserDescription());
+    }
+
+    /**
+     * Creates a new SimpleMessageSender based on the config in AppConfig
+     * @return a new SimpleMessageSender
+     */
+    private SimpleMessageSender getMessageSender() {
+        Rabbitmq config = this.appConfig.getRabbitmq();
+
+        return new SimpleMessageSender(config.getHost(), config.getPort(), config.getUsername(), config.getPassword());
+    }
+
+    /**
+     * Method to test the flow receving a message that a sample upload has finished.
+     * - Create a collection exercise
+     * - Get the collection exercise
+     * - Link the collection exercise to a sample summary with a random sample summary id
+     * - Send a message to Sample.SampleUploadFinished.binding key on sample-outbound-exchange
+     * - Get the sample links for the collection exercise
+     * - Assert the sample link is active (and the sample summary ids match)
+     *
+     * @throws CTPException throw if errors occur in any of the interactions
+     */
+    @Test
+    public void shouldActivateSampleLink() throws CTPException {
+        try {
+            String exerciseRef = "899991";
+            String userDescription = "Test Description";
+            Pair<Integer, String> result = this.client.createCollectionExercise(TEST_SURVEY_ID, exerciseRef, userDescription);
+
+            assertEquals(201, (int) result.getLeft());
+            CollectionExerciseDTO newCollex = this.client.getCollectionExercise(result.getRight());
+
+            log.info("Collection exercise to link: {}", newCollex);
+
+            UUID sampleSummaryId = UUID.randomUUID();
+
+            int status = this.client.linkSampleSummary(newCollex.getId(), sampleSummaryId);
+            assertEquals(200, status);
+
+            SimpleMessageSender sender = getMessageSender();
+
+            SampleSummaryDTO sampleSummary = new SampleSummaryDTO();
+            sampleSummary.setId(sampleSummaryId);
+
+            // This will cause an exception to be thrown as there is no collection instrument service but this is
+            // harmless to our purpose
+            sender.sendMessage("sample-outbound-exchange",
+                    "Sample.SampleUploadFinished.binding",
+                    this.mapper.writeValueAsString(sampleSummary));
+
+            Thread.sleep(500);
```

Comment:
Seems sketchy to have a Thread.sleep here. Let's change it to some kind of backing off retrying policy. https://github.com/rholder/guava-retrying is good for that sort of thing